### DISCUSSION TITLE
Serverless callback not received after sending large file for transcription

### DISCUSSION THREAD

**Discussion by jaxomlotus** [deepgram/community#174] [General help] [answered]
### Which Deepgram product are you using?

Deepgram API

### Details

I am posting to the deepgram API and waiting for a callback post with my data to my endpoint (https://transcribe.boredkatz.com/api/public/v0/broadcast/transcribe). However I am only receiving responses from short transcripts. Large transcripts never make it to my endpoint. I'm pretty sure the problem is on Deepgram's end, because I tested posting large payloads to my form via CURL and it registered. But when deepgram is making the post, I'm only ever seeing server events for short transcripts - I don't even receive an error for large payloads at all, leading me to believe it's not sending from deepgram's end.

My endpoints are hosted in AWS fargate, in case that makes a difference.

### If you are making a request to the Deepgram API, what is the full Deepgram URL you are making a request to?

https://api.deepgram.com/v1/listen?model=nova&smart_format=true&punctuate=true&utterances=true&paragraphs=true&diarize=true&callback=https://transcribe.boredkatz.com/api/public/v0/broadcast/transcribe

### If possible, please attach your code or paste it into the text box.

When I tried pasting the full code, github would not let me.  But you can get it from transcribing the mp3 file I pasted in the next form field.

### If possible, please attach an example audio file to reproduce the issue.
https://rb.gy/rognh

**Comment by lukeocodes** [comment]
hey @jaxomlotus can you confirm whether the large requests you made through curl that did work, also included a callback URL? or is it only when you use a callback URL with large audio that you don't get data sent?

**Reply by jaxomlotus** [reply]
To clarify: curl requests posting large amounts of json data to my end point directly work fine.

Callbacks from deepgram with small amounts of data work fine. 

Callbacks from deepgram with large amounts of data are never received.

**Reply by jaxomlotus** [reply]
I'm pretty sure the issue is only with the callbacks not getting sent when the data payload is large.

When I tried not using callbacks with large payloads I would get an eventual transcription response. (However I had to stop using that direct method because serverless endpoints have a 60 second timeout limitation.)

**Reply by lukeocodes** [reply]
That's for the clarification. I'm going to go out on a limb and share this: https://developers.deepgram.com/docs/wip-payload-too-large

It's an unreleased troubleshooting guide that relates to the callback server recieving a payload that is too large, and throwing an error back to our server. ~You may be able to see evidence of this in https://console.deepgram.com under usage -> logs.~  correction: you can't see this from our logs yet, we have work in progress to address this.

It would seem to fit this scenario, even without more evidence so far.

edit: if your callback server is also serverless, there will almost certainly be a request size limit. 1MB is the default on nginx, 6MB on lambda-based serverless (including Netlify, AWS Lambda)

**Comment by jaxomlotus** [comment]
We are using AWS Fargate with no max payload size.

I confirmed it’s not a payload size issue since I did CURL submits to my
server with large payloads and it was not rejected.

It’s only not being received when the callback is sent. I’m not seeing
anything in my server logs (eg no error - it’s as if it is not being
attempted). And in deepgrams logs I see a 200 success message.

**Reply by lukeocodes** [reply]
In the response you get to the initial request where the callback fails, you'll receive a small response with a request ID. Could you please share that?

**Reply by jaxomlotus** [reply]
here's a recent one that was never received with an hour long transcript:

d4588f49-2edb-4c6e-b9db-0fff027d41c7

**Reply by lukeocodes** [reply]
Gime me some time, I will attempt to find out what happened with this one.

**Reply by jaxomlotus** [reply, marked as answer]
OK, so i made some changes to my config and im seeing larger files now make it through. So I guess this was on my end after all. 

For anyone else who is using Next.js (outside of a serverless setup) and encounters this, add this to your endpoint (adjust the size limit appropriately)

```
export const config = {
  api: {
    bodyParser: {
      sizeLimit: '64mb',
    },
    responseLimit: '64mb',
  },
}
```

**Reply by lukeocodes** [reply]
Ahh bodyParser! I disabled it for another reason and manually parsed the JSON on my Next project, which uses a callback! Lucky me, I never got as far as this headscratcher... Thanks for updating us!
